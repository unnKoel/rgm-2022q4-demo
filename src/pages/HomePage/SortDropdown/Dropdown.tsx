import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  gap: 13px;

  > form {
    margin-block-end: 0; // override
    position: relative;
    > select {
      appearance: none; // override
      background-color: transparent; // override
      border: none; // override
      color: white;
      font-size: 16;
      text-transform: uppercase;
      width: 160;
      padding: 5;
      cursor: pointer;
      > option {
        background-color: #232323;
        text-transform: initial;
      }
    }
    /* Triangle */
    &::after {
      content: '';
      border: 6px solid;
      border-color: #f65261;
      border-left-color: transparent;
      border-right-color: transparent;
      border-bottom-color: transparent;
      position: absolute;
      top: 11;
      right: 4;
    }
  }
`

interface DropdownProps {
  defaultValue: string
  onChange: (value: string) => void
}

const Dropdown: React.FC<DropdownProps> = ({ defaultValue, onChange }) => {
  return (
    <Wrapper>
      {/* @ts-ignore */}
      <form onChange={(e) => onChange(e.target.value)}>
        <select name="sortBy" defaultValue={defaultValue}>
          <option value="release_date">Release Date</option>
          <option value="vote_average">Rating</option>
        </select>
      </form>
    </Wrapper>
  )
}

export default Dropdown
