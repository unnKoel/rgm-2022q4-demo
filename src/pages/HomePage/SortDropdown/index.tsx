import styled from 'styled-components'
import Dropdown from './Dropdown'

const Wrapper = styled.div`
  display: flex;
  gap: 30px;
  align-items: center;

  > .title {
    font-size: 16;
    color: white;
    opacity: 0.6;
    text-transform: uppercase;
  }
`
interface SortDropdownProps {
  defaultValue: string
  onChange: (value: string) => void
}

const SortDropdown: React.FC<SortDropdownProps> = ({
  defaultValue,
  onChange,
}) => {
  return (
    <Wrapper>
      <div className="title">Sort By</div>
      <Dropdown defaultValue={defaultValue} onChange={onChange} />
    </Wrapper>
  )
}

export default SortDropdown
