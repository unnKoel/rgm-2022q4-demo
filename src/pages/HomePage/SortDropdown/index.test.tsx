import renderer from 'react-test-renderer'
import SortDropdown from '.'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <SortDropdown defaultValue="release_date" onChange={() => undefined} />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
