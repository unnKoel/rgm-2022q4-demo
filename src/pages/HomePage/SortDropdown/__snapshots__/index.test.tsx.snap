// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="SortDropdown__Wrapper-sc-l4eavy-0 EeDhc"
>
  <div
    className="title"
  >
    Sort By
  </div>
  <div
    className="Dropdown__Wrapper-sc-1vjrvvd-0 fJVdrL"
  >
    <form
      onChange={[Function]}
    >
      <select
        defaultValue="release_date"
        name="sortBy"
      >
        <option
          value="release_date"
        >
          Release Date
        </option>
        <option
          value="vote_average"
        >
          Rating
        </option>
      </select>
    </form>
  </div>
</div>
`;
